### DISCUSSION TITLE
Thoughts on organization of work

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4190] [Feedback]
I've found myself routinely distracted by the organization of our work lately:
- the roadmap/planning has been moving around a lot and is duplicate/conflicting in several places currently
- the simple issue count is increasing (was 300 a few weeks ago (I think), now above 400)
- we have several duplicate issues and unclear issues (working on this)
- issues aren't consistently an actionable thing; many are [roadmap]s, discussions, feedback, etc.
- labels are underused (=> bad search)
- multiple label systems are in place (it's confusing)

---

When I approach an Issues list, I want to quickly be able to answer questions like "I have a surprise 30mins free - what can I get done quickly?" or "what are a few actionable user-facing Issues I can work on this month?". Instead, I currently feel a bit lost when reviewing the current Issues list. (maybe the best advice here is to manage the affect rather than the cause - I'm working on it!)

A few things would help me feel more comfortable with our organization of work:
- (in-progress) remove duplicates
- (in-progress) ask for clarity on existing issues
- move to a more consistent labeling system that should last us ages
- assign all existing issues to those labels
- all new issues are assigned a `needs-dark-review` label automatically, which we do when available
- close out non-actionable Issues; break them down into issues, move some to Discussions, some should just be labels
- keep meta/planning in one place, rather than sprawled out. The [priorities](https://github.com/orgs/darklang/projects/5) board should be plenty.

---

Idea for labeling system:

- Audience: dark, user, granduser
- Status: ready, needs thought, in progress, done
- Code (changes required): client, backend, ocaml, rust, infra, dark, documentation 
- Component: editor, editor-access, language, Cron, worker, websockets, http handler, http client, hosting, traces
- Urgency: high, not-soon
- Estimate: (I don't care what system)
- Type: bug, new-thing, improvement
- Theme: good first experience, performance, be transparent, etc. (the existing themes, I already like!) 

e.g. `audience-dark`, `code-backend`, `component-language`

---

We're certainly at a point where the backlog is more than can fit into a head at once; now that we're out of the F# port weeds, I think it's good to think about this briefly, and just stick to a system we both like.

If this seems reasonable, we can create a REPL that adds a `needs-dark-review` label, and we can review issues slowly over the next few days; that process will resolve most of my concerns. I hope this doesn't come across as "overbearing process guy" - just having a hard time right now with the current system.

**Comment by pbiggar** [comment]
I think what's happening here is different expectations for the issues and how we use them. Issues at the moment is a combination of bug reporting, project management, icebox, contributor engagement, and roadmap (and probably some more stuff). 

From a project management perspective, I guess it most closely resembles an icebox now - a place where issues go to die 😂. We write them down, we might come back to them - it's useful to have them written down and documented that we are or are not working on them, but we're not actively triaging.

It definitely has not been optimized as place to find quick issues to fix. Which isn't to say that it couldn't be or shouldn't be. I will say that my experience of trying to keep issue trackers clean in the past has not been a good one, and I think it will be hard without spending significant resources on it.

I've been using the [Priorities project](https://github.com/orgs/darklang/projects/5/views/1) as actual priorities, with the "Ready to go" column being things we could pick up quickly. I refresh that periodically by going spelunking in Issues for interesting stuff based on the themes we're working on.

The intent I had for picking what to work on in the future was to spend our time writing stuff in Dark, and fixing issues that we came upon while doing it. A nice groomed backlog isn't exactly the opposite of that, but we aren't able to do that yet and if we're going to have a big push to get to somewhere where we can set priorities and find work better, that's where I'd put my efforts, personally.

With that context: I'd suggest that we discuss what we're looking for from the system, and discuss how to get there. I think how to groom issues can be more easily solved once we're aligned on the goals.
